src/action.ts

```diff
@@ -234,7 +232,7 @@ export async function linkAllFilesInMetafile(
 		}
 		sourceRoot = searchee.path;
 	} else {
-		const downloadDirResult = await getClient().getDownloadDir(
+		const downloadDirResult = await getClient()!.getDownloadDir(
```

Comment:
didn't @mmgoodnow say no? that this wasn't necessary?

https://github.com/cross-seed/cross-seed/pull/779#pullrequestreview-2267434290

https://github.com/cross-seed/cross-seed/pull/779/commits/77617fa1a6d3843d7fdfe035f12e5a3066fb567b

Reply:
It is necessary for the blocklist branch

Reply:
Given that he discarded them and made that comment, I think a little more context and reasoning about why doing it this way would be welcome, perhaps.

Reply:
Having them discarded, and then throwing them back in another PR without a note in the PR description seems a little....I don't know, devious? belligerent? weird?

Reply:
The reason it has never been needed is because getClient wasn't being used outside of an injection context, which requires the client to exist. With blocklisting, we need to use the client but the user could be on `action: save`.

Reply:
> Having them discarded, and then throwing them back in another PR without a note in the PR description seems a little....I don't know, devious? belligerent? weird?

It wouldn't be here if it wasn't necessary.